cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -386,17 +391,165 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
+
+                    double doubleBeforeStat = (double) brokerBefore.get(doubleLoadParameter.getKafkaRebalanceStatusKey());
+                    double doubleAfterStat = (double) brokerAfter.get(doubleLoadParameter.getKafkaRebalanceStatusKey());
+                    double doubleDiff = doubleAfterStat - doubleBeforeStat;
+
+                    double[] doubleStats = {doubleBeforeStat, doubleAfterStat, doubleDiff};
+                    brokerStats.put(doubleLoadParameter.getKafkaRebalanceStatusKey(), doubleStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            doubleLoadParameter.getKafkaRebalanceStatusKey());
+                }
+
+            }
+
+            brokersStats.put(loadBeforeEntry.getKey(), brokerStats);
+        }
+
+        return brokersStats;
+
+    }
+
+    /**
+     * Converts the supplied JSONObject containing the response from the {@link CruiseControlApi#rebalance} or
+     * {@link CruiseControlApi#getUserTaskStatus} methods, into a map linking to a proposal summary map and a broker
+     * load map.
+     *
+     * @param  proposalJson The JSONObject representing the response from the Cruise Control rebalance endpoint.
+     * @return A map containing the proposal summary and broker load maps.
+     */
+    protected static Map<String, Object> processOptimizationProposal(JsonObject proposalJson) {
+
+        JsonArray brokerLoadBeforeOptimization;
+        JsonArray brokerLoadAfterOptimization;
+        if (proposalJson.containsKey(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey()) &&
+                proposalJson.containsKey(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())) {
+            brokerLoadBeforeOptimization = proposalJson
+                    .getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                    .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+            brokerLoadAfterOptimization = proposalJson
+                    .getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())
+                    .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+        } else {
+            throw new IllegalArgumentException("The rebalance optimization proposal returned by Cruise Control did not contain broker load information");
+        }
+
+        Map<Integer, Map<String, Object>> beforeAndAfterBrokerLoad = parseLoadStats(
+                brokerLoadBeforeOptimization, brokerLoadAfterOptimization);
+
+        Map<String, Object> optimizationProposal = new HashMap<>();
+        optimizationProposal.put(CruiseControlRebalanceKeys.SUMMARY.getKey(),
+                proposalJson.getJsonObject(CruiseControlRebalanceKeys.SUMMARY.getKey()).getMap());
+        optimizationProposal.put(BROKER_LOAD_KEY, beforeAndAfterBrokerLoad);
```

Comment:
@tomncooper- Hello. Here we have to create a config map and store the before/ after load in it, instead of building it in the status, right?